src/index.js

```diff
@@ -41,61 +42,82 @@ const MODES = {
   },
 };
 
-const args = process.argv.slice(2);
+const argv = parseArgs(process.argv);
 
-if(args[0] === '--version') {
+if(argv.version) {
   const version = require('../package').version;
   console.log(`horticulturalist-${version}`);
   return;
 }
 
-const mode = args.includes('--dev')   ? MODES.development :
-             args.includes('--local') ? MODES.local : MODES.medic_os;
+const mode = argv.dev   ? MODES.development :
+             argv.local ? MODES.local : MODES.medic_os;
 
-const shouldBootstrapDdoc = args.includes('--bootstrap');
+let bootstrapDdoc = argv.bootstrap;
+if (bootstrapDdoc === true) {
+  bootstrapDdoc = 'master';
+}
 
 const COUCH_URL = process.env.COUCH_URL;
 if(!COUCH_URL) throw new Error('COUCH_URL env var not set.');
 
 const DDOC = '_design/medic';
-
+const STAGED_DDOC = '_design/medic:staged';
 
 if(lockfile.exists()) {
   throw new Error(`Lock file already exists at ${lockfile.path()}.  Cannot start horticulturalising.`);
 }
 
 fs.mkdirs(mode.deployments);
 
-
 const db = new PouchDB(COUCH_URL);
 const apps = Apps(mode.start, mode.stop);
 
+info('Starting Horticulturalist');
 Promise.resolve()
-  .then(bootstrapDdoc)
-  .then(() => mode.startAppsOnStartup && startApps())
-  .then(() => db.get(DDOC))
-  .then(processDdoc)
+  // If we're bootstrapping pull down the DDOC, process and deploy it
+  .then(() => bootstrapDdoc && bootstrap())
+  .catch(fatality)
+  // If we're not boostrapping but we want to start apps do that
+  .then(() => !bootstrapDdoc && mode.startAppsOnStartup && startApps())
+  // In case there is an existing staged ddoc to be deployed deal with it
+  .then(() => db.get(STAGED_DDOC))
+  .catch(err => {
+    if (err.status !== 404) {
+      throw err;
+    }
+
+    info('No deployments to make upon boot');
+  })
+  .then(ddoc => ddoc && processDdoc(ddoc))
+  // Listen for new staged deployments and process them
   .then(() => {
     info(`Starting change feed listener at ${redact(COUCH_URL)}…`);
     db
       .changes({
         live: true,
         since: 'now',
-        doc_ids: [ DDOC ],
+        doc_ids: [ STAGED_DDOC ],
         include_docs: true,
+        attachments: true,
         timeout: false,
       })
       .on('change', change => {
-        processDdoc(change.doc)
-          .catch(fatality);
+        trace(`Change in ${STAGED_DDOC} detected`);
+        if (!change.doc._deleted) {
+            return processDdoc(change.doc).catch(fatality);
```

Comment:
I guess not no, it's just a habit I have with promises (easier to always do it than forget sometimes and be confused for hours).

I can see how it would be confusing, implying the return matters, I can drop it.